pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -52,10 +52,10 @@ func (e *ErrorWithCodes) Error() string {
 }
 
 var (
-	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId)`)
+	unauthenticatedRegexp               = regexp.MustCompile(`(?i)(Unauthorized|InvalidAuthenticationTokenTenant|Authentication failed|AuthFailure)`)
+	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(InvalidClientTokenId|SignatureDoesNotMatch|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId|AccessDenied|OperationNotAllowed|Error 403)`)
```

Comment:
`invalid character`, `InvalidAccessKeyId `,`query returned no results` and `invalid_client` are probably unauthenticated from a google lookup I did. Or did you confirm otherwise?

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html
`InvalidAccessKeyId` comes under 403 Forbidden, which is Unauthorized here.

`invalid client`, yes, it comes under authentication, will move it.

the `invalid character` and  I am not really sure about, saw somewhere it's error code 400, but confused where to put it, so I didn't move it.

```
Azure.RequestFailedException :’The specifed resource name contains invalid characters.’
Status: 400 (The specifed resource name contains invalid characters.)
ErrorCode: InvalidResourceName
```

`query returned no results`, I couldn't find any solid result. If you did, then it's fine, I will move it.

Reply:
It seems weird that `InvalidAccessKeyId` should indicate invalid permissions. It rather indicates invalid credentials, so we should match it to "unauthenticated". Potentially, the AWS documentation is just like this for maintaining backwards compatibility.

`invalid character` and `query returned no results` should both be "unauthenticated"